src/metabase/db/data_migrations.clj

```diff
@@ -172,6 +177,39 @@
                                  [:like
                                   :dashcard.visualization_settings "%\"click_link_template\":%"]]})))
 
+(defn- remove-group-id-from-mappings-by-setting-key
+   [mapping-setting-key admin-group-id]
+   ;; Intentionally get setting using `db/select-one-field` instead of `setting/get` because for some reasons
+   ;; during start-up setting/get return the default value defined in defsetting instead of value from Setting table
+   (let [mapping (try
+                  (json/parse-string (db/select-one-field :value Setting :key (name mapping-setting-key)))
+                  (catch Exception _e
+                    {}))]
+     (when-not (empty? mapping)
+       (setting/set-value-of-type!
+        :json mapping-setting-key
+        (into {}
+              (map (fn [[k v]] [k (filter #(not= admin-group-id %) v)]))
+              mapping)))))
+
+ (defmigration ^{:author "qnkhuat" :added "0.43.0"} migrate-remove-admin-from-group-mapping-if-needed
+   ;;  In the past we have a setting to disable group sync for admin group when using SSO or LDAP, but it's broken and haven't really worked (see #13820)
+   ;;  In #20991 we remove this option entirely and make sync for admin group just like a regular group.
+   ;;  But with this change we want to make sure we don't accidently add/remove adminusers we need to :
+   ;;  - for LDAP, if the `ldap-sync-admin-group` is disabled, remove all mapping for admin group
+   ;;  - for SAML, JWT remove all mapping for admin group
```

Comment:
Not a big deal, but I did some minor editing of this comment to make it a bit clearer (but feel free to edit further)

```suggestion
   ;;  But on upgrade, to make sure we don't unexpectedly begin adding or removing admin users:
   ;;  - for LDAP, if the `ldap-sync-admin-group` toggle is disabled, we remove all mapping for the admin group
   ;;  - for SAML, JWT, we remove all mapping for admin group, because they were previously never being synced
```